File: lua/PlayFab.lua
local PlayerData = {}

local function IsClientLoggedIn(player: Player): boolean
	local data = PlayerData[player]
	if not data then
		return false
	end

	local doesHaveSession = data.SessionTicket ~= nil
	local isValidSession = doesHaveSession and #data.SessionTicket > 0
	return isValidSession and data.EntityToken ~= ""
end

return {
	PlayerData = PlayerData,
	IsClientLoggedIn = IsClientLoggedIn,
}

File: src/PlayFab.ts
/**
 * Main interface to the PlayFab API wrapper
 */

interface IPlayFabPlayerData {
    SessionTicket: string
    EntityToken: string
    TokenExpiration: string
    PlayFabId: string
}

export const PlayerData = new Map<Player, IPlayFabPlayerData>()

/**
 * Returns if the player is logged in with PlayFab.
 * @param player Player instance to check
 */
export const IsClientLoggedIn = (player: Player) => {
    const data = PlayerData.get(player)
    if (!data) return false

    return data.SessionTicket !== undefined && data.SessionTicket.size() > 0 && data.EntityToken !== ''
}